functions/helloworld/test/sample.integration.http.test.js

```diff
@@ -15,25 +15,48 @@
 
 // [START functions_http_integration_test]
 const assert = require('assert');
-const Supertest = require('supertest');
-const supertest = Supertest(process.env.BASE_URL);
+const delay = require('delay');
+const execPromise = require('child-process-promise').exec;
+const path = require('path');
+const supertest = require('supertest');
+const uuid = require('uuid');
+
+const request = supertest(process.env.BASE_URL || 'http://localhost:8080');
+const cwd = path.join(__dirname, '..');
+
+let ffProc;
+
+before(async () => {
+  ffProc = execPromise(
+    `functions-framework --target=helloHttp --signature-type=http`,
+    {timeout: 1000, shell: true, cwd: cwd}
+  );
+
+  await delay(600);
+});
+
+after(async () => {
+  await ffProc;
+});
 
 it('helloHttp: should print a name', async () => {
-  await supertest
+  const name = uuid.v4();
+
+  await request
     .post('/helloHttp')
-    .send({name: 'John'})
+    .send({name: name})
     .expect(200)
     .expect(response => {
-      assert.strictEqual(response.text, 'Hello John!');
+      assert.strictEqual(response.text, `Hello ${name}!`);
     });
-});
+}).timeout(1000);
```

Comment:
Why do we need a timeout? Please comment here if we can't remove it.

Reply:
Removed. 🙂